usecase/src/main/java/org/mobilitydata/gtfsvalidator/usecase/crossvalidationusecase/stoptimesshapestrips/ValidateShapeIdReferenceInStopTime.java

```diff
@@ -0,0 +1,57 @@
+/*
+ *  Copyright (c) 2020. MobilityData IO.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package org.mobilitydata.gtfsvalidator.usecase.crossvalidationusecase.stoptimesshapestrips;
+
+import org.mobilitydata.gtfsvalidator.domain.entity.gtfs.ShapePoint;
+import org.mobilitydata.gtfsvalidator.domain.entity.gtfs.stoptimes.StopTime;
+import org.mobilitydata.gtfsvalidator.domain.entity.gtfs.trips.Trip;
+import org.mobilitydata.gtfsvalidator.domain.entity.notice.error.MissingRequiredValueNotice;
+import org.mobilitydata.gtfsvalidator.domain.entity.notice.error.ShapeIdNotFoundNotice;
+import org.mobilitydata.gtfsvalidator.usecase.port.ValidationResultRepository;
+
+import java.util.Map;
+
+/**
+ * Use case to validate that in `stop_times.txt` all records having a non-null value for field `shape_dist_travelled`
+ * refer to a record from `trips.txt` that itself refers to an existing record of `shapes.txt`.
+ */
+public class ValidateShapeIdReferenceInStopTime {
+    public void execute(final ValidationResultRepository resultRepo,
+                        final StopTime stopTime,
+                        final Map<Integer, ShapePoint> shape,
+                        final Trip trip) {
+        if (stopTime != null) {
+            if (stopTime.getShapeDistTraveled() != null) {
+                if (trip != null) {
```

Comment:
if `trip` is `null` here, shouldn't we generate another notice? If so, maybe add a `// TODO - xxxxx` statement for now? If it's already covered by another rule, I'd add a comment saying that.

Reply:
I'll add a comment to specify this, since cross validation between `trips.txt`and `stop_times.txt` on field `stop_times.trip_id` will be done in another use case. The comment will make reference to this use case